### PR TITLE
Syndicate Conspicious Weapon Expansion

### PR DESCRIPTION
**Pull request by hyperjll**
Teaser
## About The Pull Request

After about a month of on-and-off work, and after being told to atomize the work overall, here we are.
This PR will add 28 new listings to the Conspicuous Weapon tab and 16 new listings to the Ammunition tab to make the Syndicate's arsenal for traitors (and spies?) feel less barren and give traitors more options for going loud.

A full list of current additions include:

The Stinger  - 7 TC - Progression: 15m

A sword provided to high-ranking staff of the Bee Liberation Front and often seen as it's 'pride and joy'.
The sword has a 20% chance to block attacks, has high armor penetration and applies 6 units of Histamine on hit. 
Small enough to fit within a backpack.

Electrified Trident  - 9 TC - Progression: 25m

A trident rumored to have been recovered from the city of Atlantis.
It deals moderate damage, and twice as much if thrown. 
Targets attacked by this weapon will have Teslium manifested within their blood, leading to electrical shocks.

Bo Staff  - 8 TC - Progression: NONE

A wielded wooden staff that can be used to incapacitate opponents if you don't intent to harm them.
This often results in knocking your targets unconscious for a brief period of time.

Katana  - 10 TC - Progression: 25m

A brilliantly made katana capable of major damage against pretty much anyone. Has high damage, blocking ability, and inflicts wounds easily.
Doesn't do anything special other than kill, for when you just want to get the job done. Not easily concealable.

Vxtvul Hammer  - 12 TC - Progression: 30m

A relic sledgehammer with charge packs wired to two blast pads on its head.
While wielded in two hands, the user can charge a massive blow that will shatter construction, hurl bodies, and demolish mechs. Not easily concealable.

Energy Spear  - 10 TC - Progression: 30m

A spear fashioned from iron and titanium, it incorporates hardlight technology to form the pointy-end.
Turns on when wielded, and makes no noise when doing so. 
More effective against all targets than our energy swords and possesses high armor-piercing.
Deals nearby double it's damage when thrown and our hardlight technology will ensure the spear remains embedded.
Not easily concealable. Has high durability but will eventually break.

M1911  - 6 TC - Progression: NONE

A single M1911 Pistol that fires .45 bullets. Loaded with 9 lethal shots.

Viper - 8 TC - Progression: NONE

A single Viper Pistol that fires 10mm bullets. Loaded with 9 lethal shots.
This pistol fires quickly and is fully-automatic.

.357 Revolver - 9 TC - Progression: NONE

A modernized, .357 revolver. Grown straight from our gatfruit farms. Fires 7 brutal rounds of .357 magnum.
Less obvious than our modernized revolvers, for plausible deniability.

Python  - 11 TC - Progression: NONE

A single Python Revolver that fires .357 magnum bullets. Loaded with 9 brutal .357 magnum shots.

Desert Eagle  - 10 TC - Progression: NONE

A high-power Desert Eagle that fires .50 AE bullets. Loaded with 8 highly lethal shots.

Fire-y Axe  - 10 TC - Progression: 30m

A classic fire axe that's been augmented with advanced atmospheric incendiary technology.
In addition to being a fireaxe, it'll light your victims ablaze with every hit.

Energy Fire Axe  - 14 TC - Progression: 30m

A terrifying axe with a blade of pure energy, able to tear down structures with ease.
Easier to store than a standard fire axe while inactive. Has high blocking ability and armor penetration while active.

Syndicate Kammerer  - 9 TC - Progression: NONE

A shotgun that doesn't need to have it's pump racked to switch to the next shell, can have 8 shells loaded at any time.
It'll be loaded with lethal shells upon purchase.

Bolt-Action Rifle  - 3 TC - Progression: NONE

A shoddy bolt action rifle, it has 5 lethal 7.62 rounds loaded. 
It's handling is rather obnoxious and should only be used if you're on a budget.

Extended Capacity Hyper-Soaker  - 3 TC - Progression: NONE

A simple yet effective way of applying chemicals to a target's skin.
Comes with a high-power nozzle and larger tank. You'll have to provide your own chemicals, though.
(It fires 10 unit projectiles of reagents from within it's internal tank.)

Tommy Gun  - 15 TC - Progression: 25m

A functional tommy gun loaded with 50 .45 bullets. It's very much a 'spray-and-pray' kind-of weapon.
Fully automatic and surprisingly accurate. Not easily concealable.

AKM Assault Rifle  - 13 TC - Progression: 20m

A functional AKM Assault Rifle loaded with 30 7.12x82mm bullets. 
It's exceptional in combat and often reliable in taking down all kinds of opposition.
Always fires in a three-round burst. High spread.

Flamethrower (bundle?) - 8 TC - Progression: NONE

A box utilizing bluespace technology, it contains all of the equipment usually found within an incendiary weapons crate.
Contains exactly: One flamethrower, three plasma tanks, three incendiary chemical grenades.

Energy Pistol  - 7 TC - Progression: NONE

An energy pistol we've designed to look identical to that of a NT-brand disabler. 
It's capable of firing both lethal and disabling shots.
Recharges automatically.

Inferno Pistol  - 7 TC - Progression: NONE

A modified handcannon with a metamorphic reserve of decommissioned weaponized nanites.
Shoots shards of molten nanites that cause cold targets to explode.
This version has been modified to recharge automatically.

Cryogenic Pistol  - 7 TC - Progression: NONE

A modified handcannon with a metamorphic reserve of decommissioned weaponized nanites.
Shoots shards of freezing nanites that deal extra damage to hot targets, and will knock them prone.
This version has been modified to recharge automatically.

Chemical Sprayer  - 16 TC - Progression: NONE

A chemical cannon that 'sprays' 30 units to 50 units of it's contents forward for up to 5 meters (tiles) ahead.
The sprayer comes loaded with 100u's of: Sodium Thiopental, Coniine, Venom, Condensed Capsaicin, Initropidril, and Polonium.
The sprayer slowly regenerates it's chemicals using redspace technology.

Syndicate Energy Rifle  - 14 TC - Progression: NONE

A laser rifle originally manufactured by Nanotrasen, it's been augmented to recharge automatically.
Can still be recharged normally. Only fires lethal shots.

Biological Demolecularisor  - 10 TC - Progression: NONE

A gun that discharges high amounts of controlled radiation to slowly break a target into component elements. 
Deals small amounts of toxin damage (9 dmg) per shot and may irradiate targets (20% chance on hit). 
Recharges automatically but very slowly.

Mind Flayer  - 19 TC - Progression: NONE

A weapon originally created by a alien species shrouded in mystery. 
It fires energy shots that severely damage the target's nervous system and wreck havok upon the target's brain.
Also causes oxygen deprivation in the target. Recharges automatically but very slowly.
(Deals 10 OXY damage per shot, 20 BRAIN per shot, and causes hallucinations for 20 seconds after.)

Bluespace-Gravitational Manipulator  - 2 TC - Progression: NONE

An experimental, multi-mode device that fires bolts of Zero-Point Energy, causing local distortions in gravity.
Has a very slow fire-rate, and three methods of gravity manipulation including Repulse, Attract, and Chaos.

Ballistic Harpoon Gun  - 14 TC - Progression: NONE

A weapon favored by carp hunters, and infamously used by agents of the Animal Rights Consortium against human aggressors.
This speargun fires devastating harpoons which deal high amounts of damage and causes quite a lot of bleeding in the target.
Comes loaded with one harpoon. Ammunition can be purchased in bulk under the ammunition tab.

New Ammo Listings below:

10mm Handgun Magazine  - 1 TC - Progression: NONE

An additional 8-round 10mm magazine; compatible with the Viper.

10mm Armor-Piercing Magazine  - 2 TC - Progression: NONE

An additional 8-round 10mm magazine; compatible with the Viper.
These rounds are less effective at injuring the target but penetrate protective gear.

10mm Hollow-Point Magazine  - 3 TC - Progression: NONE

An additional 8-round 10mm magazine; compatible with the Viper.
These rounds are more damaging but ineffective against armour.

10mm Incendiary Magazine  - 2 TC - Progression: NONE

An additional 8-round 10mm magazine; compatible with the Viper.
Loaded with incendiary rounds which inflict reduced damage, but ignite the target.

.45mm Handgun Magazine  - 1 TC - Progression: NONE

An additional 8-round .45mm magazine, compatible with the M1911 pistol.

.45mm Armor Piercing Magazine  - 2 TC - Progression: NONE

An additional 8-round .45mm magazine, compatible with the M1911 pistol.
Exceptional when used against armored targets.

.45mm Hollow Point Magazine  - 3 TC - Progression: NONE

An additional 8-round .45mm magazine, compatible with the M1911 pistol.
Ineffective against armored targets, but very good again non-armored targets.

.45mm Incendiary Magazine  - 2 TC - Progression: NONE

An additional 8-round .45mm magazine, compatible with the M1911 pistol.
These bullets will lit your targets ablaze, though they do less damage.

.50 AE Handgun Magazine  - 4 TC - Progression: NONE

An additional 7-round .50 AE magazine, compatible with the Desert Eagle.

Stripper Clips  - 2 TC - Progression: NONE

Five stripper clips for those shoddy bolt action rifles.

Tommy Gun Drum Magazine  - 6 TC - Progression: NONE

An additional 50-round .45 caliber drum magazine, compatible with the Tommy Gun.

Rifle Magazine (7.12x82mm)  - 3 TC - Progression: NONE

An additional 30-round 7.12x82mm magazine, compatible with the AKM Assault Rifle.

Armor-Piercing Rifle Magazine (7.12x82mm)  - 4 TC - Progression: NONE

An additional 30-round Armor-Piercing 7.12x82mm magazine, compatible with the AKM Assault Rifle.

Hollow-Point Rifle Magazine (7.12x82mm)  - 3 TC - Progression: NONE

An additional 30-round Hollow-Point 7.12x82mm magazine, compatible with the AKM Assault Rifle.

Incendiary Rifle Magazine (7.12x82mm)  - 3 TC - Progression: NONE

An additional 30-round Incendiary 7.12x82mm magazine, compatible with the AKM Assault Rifle.

Harpoon Quiver  - 4 TC - Progression: NONE

A quiver containing 7 harpoons for use with the Ballistic Harpoon Gun.

## Why It's Good For The Game

Ever since playing on other codebases, i was rather shocked to find TG's uplink to be so... barren. I was told previously that we didn't NEED more syndicate items, but having a more... fleshed out uplink would lead to an increase in traitor gameplay diversity and perhaps a shift away from constant e-sword use. Do let me know if there are any balance concerns, i'm happy to adjust things or remove certain items should they be deemed undesirable.

## Changelog
:cl:
add: Added 28 new Conspicuous Weapons and 16 new Ammo purchases to the Syndicate Uplink
balance: The Syndicate Revolver, ever so prized, has been given updated technology allowing for 12 shots within it's chamber.
sound: Added a few new sounds for the new weapons.
image: Added a bunch of new icons for the new weapons.
/:cl:

### CODE REVIEW

File: code/game/objects/items/pneumaticCannon.dm [hunk 1]
```diff
@@ -337,6 +337,57 @@
 	maxWeightClass = 6 //2 pies
 	charge_ticks = 2 //4 second/pie
 
+/obj/item/pneumatic_cannon/speargun
```
Comment (necromanceranne) [inline code/game/objects/items/pneumaticCannon.dm:340]: This already exists and was changed to a bolt-action rifle. It currently is only located in a specific traitor bundle.

File: code/modules/projectiles/projectile/bullets/rifle.dm [hunk 2]
```diff
@@ -75,3 +75,32 @@
...
+	name = "7.12x82mm armor-piercing bullet"
+	damage = 35
+	wound_bonus = -45
+	armour_penetration = 45
+	bare_wound_bonus = -10
+
+/obj/projectile/bullet/mm712x82/hp
+	name = "7.12x82mm hollow-point bullet"
+	damage = 55
+	armour_penetration = -35
```
Comment (necromanceranne) [inline code/modules/projectiles/projectile/bullets/rifle.dm:98]: This indicates that this is from a very old downstream. We no longer use negative armor values. Instead, if you want to make something vulnerable to armor, you can instead do ``weak_against_armour = TRUE``. We haven't used negative values in I think like four years.

Comment (hyperjll) [reply code/modules/projectiles/projectile/bullets/rifle.dm:98]: Yeah, this has been in the works for a LONG while.

File: code/game/objects/items/pneumaticCannon.dm (hunk 1 above)
Comment (Profakos) [reply code/game/objects/items/pneumaticCannon.dm:340]: KIT_WHITE_WHALE_HOLY_GRAIL

Comment (hyperjll) [reply code/game/objects/items/pneumaticCannon.dm:340]: Ah, i found it. Will work on a fix.

**Comment by Profakos** [comment]
Trident from atlantis sounds more like a chaplain thing. Also the Stinger is very similar `/obj/item/melee/beesword`, without the rapid cooldown and with a different reagent.

**Comment by hyperjll** [comment]
> > Trident from atlantis sounds more like a chaplain thing. Also the Stinger is very similar `/obj/item/melee/beesword`, without the rapid cooldown and with a different reagent.
> 
> Yeah I just noticed some of the listed weapons, the bo stick and trident are straight up magic items that don't fit traitor role at all, either resprite them into some high tech syndicate weaponry or make them restricted to chaplain tot

I don't see how a giant wooden stick that bashes people's heads in doesn't fit the traitor role, but i can see why the trident might be seen as a magic item, likely due to the description of the item. I could just make it say it has internal electrical technology or some other sci-fi bullshit.

**Comment by hyperjll** [comment]
I'm trying my bloody hardest to figure out what the error is.

**Comment by Profakos** [comment]
The error is on `/obj/projectile/energy/radiation/on_hit`.

**Comment by Profakos** [comment]
Just further note: I have found this out because linter warnings are visible and searchable in the diff.

**Comment by hyperjll** [comment]
> Just further note: I have found this out because linter warnings are visible and searchable in the diff.

Yeah, that's what i was looking around in. I fixed one of the bugs that way, but couldn't figure out where the second one was.

**Comment by necromanceranne** [comment]
Keeping it short and blunt, this pr really doesn't sit right with me whatsoever. The actual _new_ items are maybe neat ideas, but the vast majority of this pr is a lot of reskinned versions of the same weapons or just literally crew weapons but better. Deagles, two types of revolver (?), three or so varieties of makarovs/literally just the ansem, and additionally just powercreeped versions of the novelty items from the syndicate bundles.

When approaching uplink changes, I invite you to think more towards 'what new thing could this change do for the experience that wouldn't be reasonable to include as equipment anyone could get' rather than 'Should I kill everyone with a Russian or American firearm today, maybe I'm feeling quirky'.

It also would help to atomize this PR. That is, splitting the various bits and pieces into separate prs. This is so much stuff and addressing the individual aspects and problems of the PR is going to get very crowded. Some parts of it needs their own discussions to address, particularly with regards to code quality.

**Comment by hyperjll** [comment]
> Keeping it short and blunt, this pr really doesn't sit right with me whatsoever. The actual _new_ items are maybe neat ideas, but the vast majority of this pr is a lot of reskinned versions of the same weapons or just literally crew weapons/equipment but better. Deagles, two types of revolver (one just has near double the ammo), three or so varieties of makarovs/literally just the ansem, and additionally just powercreeped versions of the novelty items from the syndicate bundles.
> 
> When approaching uplink changes, I invite you to think more towards 'what new thing could this change do for the experience that wouldn't be reasonable to include as equipment anyone could get' rather than 'Should I kill everyone with a Russian or American firearm today, maybe I'm feeling quirky'.
> 
> It also would help to atomize this PR. That is, splitting the various bits and pieces into separate prs. This is so much stuff and addressing the individual aspects and problems of the PR is going to get very crowded. Some parts of it needs their own discussions to address, particularly with regards to code quality.

By all means, give me some examples to work with, and make some suggested changes. If you don't approve of any particular item, i'm willing to scrap it provided good reason. (I worked for nearly a month on this project, i'm not super willing to toss things out because it just doesn't sit right with you.)

**Comment by starrm4nn** [comment]
isn't the tommy gun an admin weapon that insta-stuns/knocks down on hit

**Comment by hyperjll** [comment]
> isn't the tommy gun an admin weapon that insta-stuns/knocks down on hit

No.

**Comment by necromanceranne** [comment]
I mean, the first suggestion I'd make is that anything that is analogous to another traitor item, a nukie item (I'm looking at that tommy gun and the 10mm pistol), a crew item (you can count the m1911 as crew gear now that we have a role that spawns with it) or the exclusive bundle stuff should go or be reintroduced in some other way. The reason is not only redundancy is just simply not necessary, but their inclusion diminishes the aspect of the sandbox that matters. Interacting with the map and possibly players to acquire equipment to utilize as part of your traitoring.

If, for example, you wanted to introduce a special traitor version of the fireaxe, you could do what I did with snatcherprods. Introduce some kind of crafting recipe that requires you to still get the various parts, but still need traitor specific items (like telecrystals) to actually make. It removes some clutter, but makes for a little bit of a hidden weapon people can seek out, and forces players to engage with the sandbox without simply stepping out of maint with all the gear they need to murderize with impunity. 

Different flavours of the same murder tool is just not an interesting inclusion is maybe my biggest piece of advice.

**Comment by Time-Green** [comment]
This is all just more for the sake of more. I also strongly disagree with your “Why it’s good for the game”. 

> Ever since playing on other codebases, i was rather shocked to find TG's uplink to be so... barren. I was told previously that we didn't NEED more syndicate items, but having a more... fleshed out uplink would lead to an increase in traitor gameplay diversity and perhaps a shift away from constant e-sword use. Do let me know if there are any balance concerns, i'm happy to adjust things or remove certain items should they be deemed undesirable.

It wont make gameplay more diverse, it’s mostly the same stuff that’s already in the uplink or elsewhere, but slightly different (fire-y axe????). If these were unique items adding cool new playstyles, hell yeah! But it’s just a lot of them to hide the fact that they aren’t interesting in isolation 

There’s also some items that really shouldn’t be in there at all, like the grav cannon or flamethrower. Stuff like assault rifles is also more for nukie gameplay, and they got their own versions. 

More is not better, this is just a lot of fats. If anything, players are gonna be overwhelmed by the 30 slightly different new guns and melee options that all end up doing the same (murder). Choice overload is a real thing. This just leads to more braindead gameplay in the end

Some of these are okay as one time funny items, like the trident and energy spear, but not in the permanent uplink roster. I would be okay with implementation elsewhere or as some sort of rare uplink item

**Comment by Time-Green** [comment]
I know you spent a lot of time on this, but you really should talk with us before you spent a lot of time coding something 

I suggest you don’t add or remove anything for a bit so you don’t run around following request after request just for it to get completely shut down

I’m putting it on DNM until some other maintainers are willing to weigh in

**Comment by Rex9001** [comment]
Perhaps this pr could be atomised to seperate the wheat from the chaff so to speak

**Comment by ZephyrTFA** [comment]
This isn't what atomization means. This is about as un-atomized as you can get.
Atomization would be adding these in distinct sets, letting them get balanced, and adding more afterward.
I'm sorry to do this to you.

**Comment by ZephyrTFA** [comment]
Don't take what I said to mean that I/we don't want this.
It's cool as fuck.
But you need to split it up!

**Comment by Exester509** [comment]
I'm with Time-Green on this, most of these seem bland or not fitting for tgstation, or outright reskins of things we have. We really don't need three .357 revolvers and two 10mm pistols that just do the same thing.

The egun disguised as a disabler and the water gun seem like very interesting concepts though. (Also, some of these don't even work as you'd expect with tg's implementation of things like radiation and chemical spraying)

**Comment by EricZilla** [comment]
I recommend doing maybe four weapons at a time so its not a huge amount of feedback all over the place and not a huge pain for the maint team to sift through for QC

**Comment by Sadboysuss** [comment]
Zero-point gravity gun is already an item available to the science team at later stages of the game

**Comment by SomeoneYouProbablyKnow** [comment]
My problem is that a lot of those items are very redundant 

> The Stinger
Electrified Trident
Katana
Vxtvul Hammer
Energy Fire Axe

Those are all "hit em till they die" weapons. How are those actually different from an esword or a desword (whichever is closest in price). They do look neat but why not make them into weapon skins.

> Energy Spear 

The durability aspect is neat something can be done with that i'm sure.

> Fire-y Axe 

Funny pun and a sort of unique effect

> Syndicate Kammerer

Since we don't have non-syndicate Kammerers the name doesn't make sense. But you shouldn't use Kammerer's name on an evil gun the reference doesn't make sense.

>M1911 
Viper
.357 Revolver - 9 TC - Progression: NONE

Guh? we have that already for 13 TC don't we. The prices are all messed up but that's not my problem

> Python
Desert Eagle
Bolt-Action Rifle
Tommy Gun 
AKM Assault Rifle

These are just guns. You can already buy or get access to guns as a syndi. Maybe one really high end gun makes sense but there is no point in having so many.

> Flamethrower

Neat i guess. Those are easy to make but what is the uplink for if not for saving on time and giving people ideas.

> Energy Pistol
Inferno Pistol
Cryogenic Pistol
Syndicate Energy Rifle
Biological Demolecularisor
Mind Flayer

For all the bells and whistles those are just recharging guns. 

> Bluespace-Gravitational Manipulator

This one is actually unique and neat.

**Comment by Jacquerel** [comment]
Yeah I am afraid that in our conversation yesterday we did not get sufficiently onto the same page.
You need to make one PR _**per item**_, or per "very similar" item.
This PR would be impossible to merge because an objection to one item (and there would be many) blocks the entire PR.

Each PR needs to explain _what unfilled niche this item is serving_.
"I added a new item because more is good" is insufficient.
We already have a lot of traitor items, and we don't need or really actually want there to be four or five different very close variants of the same thing in there.

Currently the weapons in the traitor uplink are like:
```
Baton- Instantly win any duel
ESword- A well-rounded melee item that is very threatening and sort of defensive, can work as an offhand shield
Dual Esword- Go all in, in exchange for greater defence
EDagger- Cheap and concealable backup weapon
Powerfist- Mostly a gimmick for pushing people into things, or a risky means of inflicting very large chunks of damage
Sleeping Carp- Dedicate yourself to the way of melee to an even greater extent than the eblade to turn your body into a weapon

EBow- "nonlethal" concealable ranged tool for catching or locking down one or two people
Revolver- The big gun which kills people loudly and effectively
Makarov - Cheaper, more easily concealed
```
This doesn't leave a lot of unfilled needs for you to slot into, and as I said we kind of don't need more just for the sake of having a bigger list to scroll through.

I will be honest I think it is going to be kind of hard to convince me of the necessity of... quite a large number of these. Not even mentioning the ones which are basically just magical artefacts.
Like... a disabler? A laser gun? A _flamethrower_? What are those even on there for?

And also when these things are taken from other codebases you should credit them directly, the specfici person who added it when possible.

I understand you spent a month of coding on doing traitor items... in silence.
The unfortunate fact is, you probably should have been talking about adding these _before_ and _during_ that month not only after 😓 
And also rather than spending 30 days on making one PR, you should have been spending eight hours per item on making 40 PRs (which is advice we could have given you if you had been talking to anyone about this).

**Comment by hyperjll** [comment]
I hear you loud and clear. But, I'm not super interested in spending more time shrinking it all down. I MAY PR what ya'll believe to be the more interesting stuff, but I'm done for now.

All in all, seems a bit like my ideas just weren't 'unique' enough to be added, oh well. I tried and that's what matters.

**Comment by Sadboysuss** [comment]
> I hear you loud and clear. But, I'm not super interested in spending more time shrinking it all down. I MAY PR what ya'll believe to be the more interesting stuff, but I'm done for now.
> 
> All in all, seems a bit like my ideas just weren't 'unique' enough to be added, oh well. I tried and that's what matters.

there's plenty of cool content to be salvaged here, don't let the hate get to you and learn from the mistakes. I hope you come back to this and make it all better, I believe in you.

**Comment by optimumtact** [comment]
also make sure you have positive GBP if you're gonna make big feature PR's

**Comment by MrDoomBringer** [comment]
do people just not know what a design doc is

**Comment by hyperjll** [comment]
Oh, i did. I'm no longer planning on contributing to TG anymore. I've since made a downstream where i'll keep my creative freedom and will have all of my features there inevitably.

**Comment by hyperjll** [comment]
> with blackjack and hookers

Maybe. :)

**Comment by AMyriad** [comment]
gotta love how you yoinked a quarter of this from yogs and gave zero credit to the creators

**Pull request closed without merging**